File: Programming-Basics-With-JavaScript/Homeworks-And-Labs/03.Conditional-Statements-Advanced/02-exercise/08.onTimeForTheExam.js
function onTimeForTheExam(input) {

    const hourOfExam = Number(input[0]);
    const minutesOfTheExam = Number(input[1]);
    const hourOfArrival = Number(input[2]);
    const minutesOfArrival = Number(input[3]);

    const examTimeInMin = hourOfExam * 60 + minutesOfTheExam;
    const arrivalTimeInMin = hourOfArrival * 60 + minutesOfArrival;
    const difference = Math.abs(examTimeInMin - arrivalTimeInMin);
    const hours = Math.floor(difference / 60);
    const minutes = difference % 60;

    if (arrivalTimeInMin > examTimeInMin) {
        console.log("Late");
        if (difference < 60) {
            console.log(`${minutes} minutes after the start`);
        } else if (difference >= 60) {
            if (minutes < 10) {
                console.log(`${hours}:0${minutes} hours after the start`);
            } else {
                console.log(`${hours}:${minutes} hours after the start`);
            }
        }
    } else if (arrivalTimeInMin === examTimeInMin || difference <= 30) {
        console.log("On time");
        console.log(`${minutes} minutes before the start`);
    } else if (difference > 30) {
        console.log("Early");
        if (difference < 60) {
            console.log(`${minutes} minutes before the start`);
        } else if (difference >= 60) {
            if (minutes < 10) {
                console.log(`${hours}:0${minutes} hours before the start`);
            } else {
                console.log(`${hours}:${minutes} hours before the start`);
            }
        }
    }
}
onTimeForTheExam(["11", "30", "10", "55"]);

File: Python_Basics/Homeworks-And-Labs/03.conditional_statements_advanced/02.exercise/08.on_time_for_the_exam.py
hourOfExam = int(input())
minutesOfTheExam = int(input())
hourOfArrival = int(input())
minutesOfArrival = int(input())

examTimeInMin = hourOfExam * 60 + minutesOfTheExam
arrivalTimeInMin = hourOfArrival * 60 + minutesOfArrival
difference = abs(examTimeInMin - arrivalTimeInMin)
hours = difference // 60
minutes = difference % 60

if arrivalTimeInMin > examTimeInMin:
    print("Late")
    if difference < 60:
        print(f'{minutes} minutes after the start')
    elif difference >= 60:
        if minutes < 10:
            print(f'{hours}:0{minutes} hours after the start')
        else:
            print(f'{hours}:{minutes} hours after the start')
elif arrivalTimeInMin == examTimeInMin or difference <= 30:
    print("On time")
    print(f'{minutes} minutes before the start')
elif difference > 30:
    print("Early")
    if difference < 60:
        print(f'{minutes} minutes before the start')
    elif difference >= 60:
        if minutes < 10:
            print(f'{hours}:0{minutes} hours before the start')
        else:
            print(f'{hours}:{minutes} hours before the start')
